docs/2.guide/2.directory-structure/1.components.md

```diff
@@ -119,6 +119,79 @@ const show = ref(false)
 </template>
 ```
 
+## Delayed Hydration
+
+In real world applications, some pages may include a lot of content and a lot of components, and most of the time not all of them need to be interactive as soon as the page is loaded. Having them all load eagerly can negatively impact performance and increase bundle size.
+
+In order to optimize the page, you may want to delay the hydration of some components until they're visible, or until the browser is done with more important tasks for example. Nuxt has first class support for delayed hydration components without having to write all of the boilerplate.
+
+In order to use delayed hydration, you first need to enable it in your experimental config in `nuxt.config`
+
+```ts [nuxt.config.{ts,js}]
+export default defineNuxtConfig({
+  experimental: {
+    delayedHydration: true
+  }
+})
+```
+
+Nuxt has reserved component prefixes that will handle this delayed hydration for you, that extend dynamic imports. By prefixing your component with `LazyVisible`, Nuxt will automatically handle your component and delay its hydration until it will be on screen.
+
+```vue [pages/index.vue]
+<template>
+  <div>
+    <LazyVisibleMyComponent />
+  </div>
+</template>
+```
+
+If you need the component to load as soon as possible, but not block the critical rendering path, you can use the `LazyIdle` prefix, which would handle your component's hydration whenever the browser goes idle.
+
+```vue [pages/index.vue]
+<template>
+  <div>
+    <LazyIdleMyComponent />
+  </div>
+</template>
+```
+
+If you would like the component to load after certain events occur, like a click or a mouse over, you can use the `LazyEvent` prefix, which would only trigger the hydration when those events occur.
+
+```vue [pages/index.vue]
+<template>
+  <div>
+    <LazyEventMyComponent />
+  </div>
+</template>
+```
+
+### Custom hydration triggers
+
+If you would like to override the default hydration triggers when dealing with delayed hydration, like changing the timeout, the options for the intersection observer, or the events to trigger the hydration, you can do so by supplying a `loader` prop to your lazy components.
+
+```vue [pages/index.vue]
+<template>
+  <div>
+    <LazyIdleMyComponent :loader="createIdleLoader({timeout: 3000})"/>
```

Comment:
The `loader` prop only makes sense if we're passing in a promise, which would only make sense for a generic component. 

```ts
<LazyMyComponent :loader="createIdleLoader({timeout: 3000})"/>
```

If you prefer not to create a low-level implementation, that's okay, but I think it's probably better to use a different prop name and the function wouldn't be required.

Reply:
I was thinking about it, but since there are defaults for all the components the loader prop is there to allow overrides. Should I name it differently?

Reply:
Yes renaming, keep in mind that `loader` is a prop for defineAsyncComponent in Vue which is a promise, we want to remove ambiguity around what the prop does

Reply:
I suggested using `loader` with the thinking that the user would be providing a promise so the signature would match